sshd-core/src/main/java/org/apache/sshd/client/config/hosts/DefaultConfigFileHostEntryResolver.java

```diff
@@ -75,12 +90,105 @@ public final boolean isStrict() {
                 log.debug("reloadHostConfigEntries({}@{}:{}) check permissions of {}", username, host, port, path);
             }
 
-            PosixFilePermission violation = KeyUtils.validateStrictKeyFilePermissions(path);
+            PosixFilePermission violation = validateStrictConfigFilePermissions(path);
             if (violation != null) {
                 throw new IOException("String permission violation (" + violation + ") for " + path);
             }
+
+            String ownerViolation = validateStrictConfigFileOwner(path);
+            if (ownerViolation != null) {
+                throw new IOException("String owner violation (" + ownerViolation + ") for " + path);
+            }
         }
 
         return super.reloadHostConfigEntries(path, host, port, username);
```

Comment:
I prefer:
```java
if (GenericUtils.length(owner) > 0) {
    throw ...
}
```
The assumption is that there is not such thing as "empty" owner - and if there is, then we assume no violation since we cannot determine the user.